### ISSUE TITLE
Test new Galil driver

### ISSUE THREAD

**Issue by FreddieAkeroyd** [ISISComputingGroup/IBEX#2830]
The new Galil driver #2151 needs testing on a beamline

**Comment by kjwoodsISIS** [comment]
Test on IMAT during shutdown (January).
Check homing inn the office first.

**Comment by AdrianPotter** [comment]
Decide on suitable testing strategy first

**Comment by AdrianPotter** [comment]
As per #2151, there needs to be a step in the upgrade script to change the argument list in GalilStartController to reflect the new interface.

**Comment by Tom-Willemsen** [comment]
After discussing with @FreddieAkeroyd , marking this as impeded until we can put the new driver on EMMA to test it (EMMA is currently being used to do runs). Probably best to do this ticket at the beginning of a shutdown.

**Comment by DominicOram** [comment]
https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/New-Galil-Driver contains a test strategy, can someone please make sure it sounds sensible?

**Comment by DominicOram** [comment]
@Tom-Willemsen and I made a start on this last shutdown but didn't get round to finishing it. I won't be around next shutdown to look at it. I've made a build of the new driver at http://epics-jenkins.isis.rl.ac.uk/job/new_galil_driver/ (however, it is worth noting that this is also now a bit out of date so we may want to update it first). Tom and I took a number of measurements in preparation of updating the driver using the following:
`
for i in range(1,9):
    print("Motor: MTR010{}".format(i))
    print("Motor pos is {}".format(g.get_pv("MOT:MTR010{}.RMP".format(i), is_local=True)))
    print("Enc pos is {}".format(g.get_pv("MOT:MTR010{}.REP".format(i), is_local=True)))
    print("")
    print("")
` 
We found that for MTR0101:

> At home:
   Motor: MTR0101
   Motor pos is 612000
   Enc pos is 153000

   At high limit:
   Motor: MTR0101
   Motor pos is 615281
   Enc pos is 153820

   At low limit:
   Motor: MTR0101
   Motor pos is 1957
   Enc pos is 503

It took ~2 seconds to from forward limit to home and ~5 minutes to go between limits. The chopper lifter took 1:40 mins for full travel.

**Comment by DominicOram** [comment]
Testing went well. I've added some notes on the parts that failed at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/New-Galil-Driver. The plan going forward is to:
* Merge master into https://github.com/ISISComputingGroup/EPICS-galil/tree/update_to_3_4 @FreddieAkeroyd to look at this
* Fix homing on datum
* Fix WLP
* Create a full VS2017 build 
* Deploy and retest on EMMA
* Look into creating a VS2017 release to deploy on a simple instrument (i.e. EMMA or ALF) before the end of cycle

**Comment by FreddieAkeroyd** [comment]
3_4 branch updated to latest upstream, our own master merged, it builds in support but not in ioc at the moment

**Comment by FreddieAkeroyd** [comment]
IOC now looks to be building OK in "new_galil_driver" Jenkins job

**Comment by FreddieAkeroyd** [comment]
Putting into impeded until further testing discussed with @DominicOram The new_galil_driver jenkins job is currently run on a VS2010 build machine but can easily be moved to a VS2017 build machine

**Comment by FreddieAkeroyd** [comment]
Test after #5031 so using latest asyn

**Comment by John-Holt-Tessella** [comment]
Can't do this until back in the office moved from main board

**Comment by DominicOram** [comment]
Aim to test on a small number of beamlines for cycle 2021/01

**Comment by FreddieAkeroyd** [comment]
Bring up to date with latest code from github first

**Comment by DominicOram** [comment]
Email sent to IS to see when a good time would be

**Comment by DominicOram** [comment]
IS says any time this shutdown is fine but to let them know when we have a better idea of dates. JN also asked that we inform them when we're doing the test in case of any electrical/mechanical issues

**Comment by DominicOram** [comment]
In meeting between @FreddieAkeroyd and myself we decided that to stop this becoming such an issue in the future as part of this ticket we should:
* Fork our galil repo from the "vendor" one
* Cherry-pick our current changes onto new feature branches with the aim to get these merged into the vendor code (aiming to keep history)
* Going forward we will point our build at the master of this fork, which will be the same as the upstream master. If we diverge we can temporarily point our build at a different branch but we should aim to keep the two masters in step as much as possible

**Comment by DominicOram** [comment]
Spun off https://github.com/ISISComputingGroup/IBEX/issues/6255 and will leave this as just the testing

**Comment by DominicOram** [comment]
Impeded until https://github.com/ISISComputingGroup/IBEX/issues/6255 is reviewed

**Comment by KathrynBaker** [comment]
This has been installed and is being tested on CRISP imminently!

**Comment by FreddieAkeroyd** [comment]
Changes to the new galil driver code is currently being built into an ibex11 branch (FreddieBuild11) for deployment to CRISP

**Comment by KathrynBaker** [comment]
@FreddieAkeroyd were these tested yet? We didn't get to look at the impeded items at sprint changeover

**Comment by FreddieAkeroyd** [comment]
I was doing some testing on CRISP, but then it was closed up for potential beam that I think didn't happen, so I need to get back to Rob to see if it is OK to go again. But it seemed fine, I wonder if EMMA is worth a go next.

**Comment by FreddieAkeroyd** [comment]
Now #8979
